src/node_http_parser.cc

```diff
@@ -462,19 +458,14 @@ class Parser : public AsyncWrap, public StreamListener {
     if (!cb->IsFunction())
       return 0;
 
-    // We came from consumed stream
-    if (current_buffer_.IsEmpty()) {
-      // Make sure Buffer will be in parent HandleScope
-      current_buffer_ = scope.Escape(Buffer::Copy(
-          env()->isolate(),
-          current_buffer_data_,
-          current_buffer_len_).ToLocalChecked());
-    }
+    // Make sure Buffer will be in parent HandleScope
+    Local<Object> current_buffer = scope.Escape(
+        Buffer::Copy(env()->isolate(), at, length).ToLocalChecked());
 
     Local<Value> argv[3] = {
-        current_buffer_,
+        current_buffer,
         Integer::NewFromUnsigned(
-            env()->isolate(), static_cast<uint32_t>(at - current_buffer_data_)),
+            env()->isolate(), 0),
         Integer::NewFromUnsigned(env()->isolate(), length)};
```

Comment:
I'd remove these (from lib/_http_common.js too) since they're not that useful anymore.

I'm 95% sure `length` is always > 0 but you can guard the call into JS to be sure. It can be simplified to this:
```c++
if (length > 0) {
  Local<Value> arg = Buffer::Copy(env()->isolate(), at, length).ToLocalChecked();
  MaybeLocal<Value> r = MakeCallback(cb.As<Function>(), 1, &arg);
  // ...
}
```